src/xrc/src/api.rs

```diff
@@ -736,11 +766,12 @@ fn handle_fiat_pair(
 }
 
 async fn get_stablecoin_rate(
+    exchanges: &[&Exchange],
     symbol: &str,
     timestamp: u64,
-) -> Result<QueriedExchangeRate, CallExchangeError> {
+) -> Result<QueriedExchangeRateWithFailedExchanges, CallExchangeError> {
     let mut futures = vec![];
-    EXCHANGES.iter().for_each(|exchange| {
+    exchanges.iter().for_each(|exchange| {
         if !cfg!(feature = "ipv4-support") && !exchange.supports_ipv6() {
             return;
         }
```

Comment:
Remove this chunk. It is unnecessary as `is_available()` check will be move into the `handle_` functions.